test/init_tracer.js

```diff
@@ -156,4 +155,26 @@ describe('initTracer', () => {
         assert.equal(tracer._metrics._factory, metrics);
         assert.equal(tracer._tags['x'], 'y');
     });
+
+    describe("upsampleChildSpans", () => {
```

Comment:
we are getting to a situation where the possible state machine is growing due to various dimensions, so we need to have a data driven test. Dimensions include (could be more, some may not be fully orthogonal):
  * root span or not
  * first in process span or not
  * child span or not
  * upstream span was sampled or not
  * upsampling enabled or not
    * upsampling on inner spans enabled or not
  * finalized or not
  * operation name changed or not
  * TODO: there's PR in OT to allow changing service name via "service" tag, which has similar effect to changing op name